src/Language/Pirouette/Example/IsUnity.hs

```diff
@@ -93,19 +59,11 @@ pairEq @a @b eqA eqB x y =
     (\(x0 : a) (x1 : b) . match_Pair @a @b y @Bool
       (\(y0 : a) (y1 : b) . and (eqA x0 y0) (eqB x1 y1)))
 
-data Maybe x
-  = Just : x -> Maybe x
-  | Nothing : Maybe x
-
-maybeSum : forall x . Maybe x -> Maybe x -> Maybe x
-maybeSum @x mx my =
-  match_Maybe @x mx @(Maybe x)
-    (\(jx : x) . Just @x jx)
-    my
-
-isJust : forall x . Maybe x -> Bool
-isJust @x mx =
-  match_Maybe @x mx @Bool (\(jx : x) . True) False
+maybeSum : forall a . Maybe a -> Maybe a -> Maybe a
```

Comment:
This looks useful! Should this go to the stdlib as well?

Reply:
I was doing things more on a define-by-need basis. But it's true that it looks rather generic. https://github.com/tweag/pirouette/pull/168/commits/0e079cbdd2715e920ad6378fd9480535e5812543 moves it to the standard library under the name `firstJust` (the only convincing name Hoogle threw at me when I asked for the type; I'm trying to stay as close as possible to Haskell type and function names).